src/psyclone/psyGen.py

```diff
@@ -3220,17 +3313,24 @@ def gen_code(self, parent):
             calls = self.reductions()
             zero_reduction_variables(calls, parent)
 
-        if self.root.opencl or (self._start == "1" and self._stop == "1"):
+        if self.root.opencl or (self.start_expr == Literal("1") and
+                                self.stop_expr == Literal("1")):
             # no need for a loop
-            for child in self.children:
+            for child in self.loop_body:
                 child.gen_code(parent)
         else:
+            from psyclone.psyir.backend.fortran import FortranWriter
             from psyclone.f2pygen import DoGen, DeclGen
-            do = DoGen(parent, self._variable_name, self._start, self._stop)
+            # Start_expr and stop_expr are generated with the FortranWriter
+            # backend, the rest of the loop with fpygen.
+            fwriter = FortranWriter()
+            do = DoGen(parent, self._variable_name,
+                       fwriter(self.start_expr),
+                       fwriter(self.stop_expr))
```

Comment:
What about the step or is that only handled by sub-classes?

Reply:
It wasn't on the original code :), but yes I added it